python/ray/dag/compiled_dag_node.py

```diff
@@ -479,6 +479,9 @@ def teardown(self):
         """Teardown and cancel all worker tasks for this DAG."""
         self._monitor.teardown()
 
+    def __del__(self):
+        self.teardown()
```

Comment:
The caller can still explicitly call it? Or do you mean we shouldn't do it implicitly at all? I put this here to avoid leaks.

Reply:
Hmm actually looking at teardown method, it doesn't use any stdlib. So I guess it should be fine. I was talking about this kind of case

```
import os

class A:
    def __del__(self):
        # os can be deallocated when this is called
         print(os)
```

Reply:
I think if there's a way we can clean it without relying on `__del__` (because upon shutdown the ordering of deallocation is not guaranteed), that's the best, but I guess it should be fine for now